src/hyperkit.c

```diff
@@ -745,51 +745,47 @@ firmware_parse(const char *opt) {
 		goto fail;
 	}
 
-		if ((cp = strchr(fw, ',')) != NULL) {
-			*cp = '\0';
-			opt1 = cp + 1;
-		} else {
-			goto fail;
-		}
+	if ((cp = strchr(fw, ',')) != NULL) {
+		*cp = '\0';
+		opt1 = cp + 1;
+	}
 
-		if ((cp = strchr(opt1, ',')) != NULL) {
-			*cp = '\0';
-			opt2 = cp + 1;
-		} else {
-			goto fail;
-		}
+	if (opt1 && (cp = strchr(opt1, ',')) != NULL) {
+		*cp = '\0';
+		opt2 = cp + 1;
+	}
 
-		if ((cp = strchr(opt2, ',')) != NULL) {
-			*cp = '\0';
-			opt3 = cp + 1;
-		} else {
-			goto fail;
-		}
+	if (opt2 && (cp = strchr(opt2, ',')) != NULL) {
+		*cp = '\0';
+		opt3 = cp + 1;
+	}
 
-		opt2 = strlen(opt2) ? opt2 : NULL;
-		opt3 = strlen(opt3) ? opt3 : NULL;
+	opt1 = opt1 && strlen(opt1) ? opt1 : NULL;
+	opt2 = opt2 && strlen(opt2) ? opt2 : NULL;
+	opt3 = opt3 && strlen(opt3) ? opt3 : NULL;
```

Comment:
I'd replace `strlen(optN)` with `*optN` in each case, we don't need to know the full length, just that it is non-zero.

At first I thought this was a NOP, until I realised it would convert `""` into `NULL`. Perhaps that is worth a comment.